lumicks/pylake/detail/h5_helper.py

```diff
@@ -2,42 +2,97 @@
 from fnmatch import fnmatch
 
 
-def write_h5(h5_file, output_filename, compression_level=5, omit_data=None):
+def write_cropped_metadata(lk_file, out_file, name, node, crop_time_range, verbose):
+    """Write non-numerical data"""
+
+    def write_node():
+        out_file.create_dataset(name, data=node)
+        out_file[name].attrs.update(node.attrs)
+
+    if not crop_time_range:
+        write_node()
+    else:
+        # Override time ranges. Items know how to crop themselves.
+        try:
+            start, stop = (
+                getattr(lk_file[name][slice(crop_time_range[0], crop_time_range[1])], field)
+                for field in ("start", "stop")
+            )
+            if stop >= crop_time_range[0] and start < crop_time_range[1] and (stop - start) > 0:
+                write_node()
+                out_file[name].attrs["Start time (ns)"] = start
+                out_file[name].attrs["Stop time (ns)"] = stop
+            else:
+                if verbose:
+                    print(f"{name} removed from file (out of cropping range)")
+        except (IndexError, TypeError):
+            if verbose:
+                print(f"{name} not cropped")
+
+
+def write_h5(
+    lk_file,
+    output_filename,
+    compression_level=5,
+    omit_data=None,
+    *,
+    crop_time_range=None,
+    verbose=False,
+):
     """Write a modified h5 file to disk.
 
     Parameters
     ----------
-    h5_file : h5py.File
-        loaded h5 file
+    lk_file : lk.File
```

Comment:
isn't renaming positional arguments breaking?

Reply:
It is, but this is private API.